Mathlib/Algebra/Polynomial/AlgebraMap.lean

```diff
@@ -615,11 +615,13 @@ lemma aeval_apply_smul_mem_of_le_comap'
     [Semiring A] [Algebra R A] [Module A M] [IsScalarTower R A M] (hm : m ∈ q) (p : R[X]) (a : A)
     (hq : q ≤ q.comap (Algebra.lsmul R R M a)) :
     aeval a p • m ∈ q := by
-  refine p.induction_on (motive := fun f ↦ aeval a f • m ∈ q) (by simpa) (fun f₁ f₂ h₁ h₂ ↦ ?_)
-    (fun n t hmq ↦ ?_)
-  · simp_rw [map_add, add_smul]
+  induction p using Polynomial.induction_on with
+  | C a => simpa using SMulMemClass.smul_mem a hm
```

Comment:
So you need to specify `SMulMemClass.smul_mem` because you don't specify the motive, right? Seems fair.

Reply:
No, previously we ran `simp` on the goal `∀ r : R, (aeval a) (C r) • m ∈ q` , which allowed it to use `SetLike.forall_smul_mem_iff`. But now the `r` is introduced automatically, so either we'd need to `revert` it or work around it (which was easy enough).